site/viamguides/src/control-module/control-module.md

```diff
@@ -0,0 +1,577 @@
+author: Joyce
+id: control-module
+summary: Add control logic to your machine to run in a loop in the background
+categories: Getting-Started, Developer
+environments: web
+status: Published
+feedback link: https://github.com/viam-devrel/viamcodelabs/issues
+tags: Getting Started, Developer
+
+# Build a Viam module for control logic
+
+<!-- ------------------------ -->
+
+## Overview
+
+Duration: 1
+
+Let’s say you have a machine in Viam with a sensor and an actuator, like a webcam and a lamp plugged into a smart outlet. You want the machine to monitor the video feed, in this case, turning on the lamp when a person is detected.
+
+To do that, you'll write a control module that checks sensor data and triggers actions automatically.
+
+![control logic workflow](assets/workflow.png)
```

Comment:
It would also be helpful to show the sensing->control->actuation (or sensor->control->actuator) flow for building a robot, from the plant watering workshop, here. It could help folks understand this module as a part of building a real-world robot.